src/libs/Formula.ts

```diff
@@ -494,20 +533,45 @@ function getSubstring(value: string, args: string[]): string {
 
 /**
  * Format an amount value
+ * @returns The formatted amount string, or null if currency conversion is needed (unavailable on frontend)
  */
-function formatAmount(amount: number | undefined, currency: string | undefined): string {
+function formatAmount(amount: number | undefined, currency: string | undefined, displayCurrency?: string): string | null {
     if (amount === undefined) {
         return '';
     }
 
     const absoluteAmount = Math.abs(amount);
-    const formattedAmount = (absoluteAmount / 100).toFixed(2);
 
-    if (currency) {
-        return `${currency}${formattedAmount}`;
-    }
+    try {
+        if (displayCurrency?.trim().toLowerCase() === 'nosymbol') {
+            return convertToDisplayStringWithoutCurrency(absoluteAmount, currency);
+        }
+
+        // Check if format is a valid currency code (e.g., USD, EUR, eur)
+        const currencyCode = displayCurrency?.trim().toUpperCase();
```

Comment:
```suggestion
        const trimmedDisplayCurrency = displayCurrency?.trim().toUpperCase();
```

We can also move this up and use it in the above condition